src/json/to_json.cr

```diff
@@ -124,7 +124,7 @@ end
 
 struct Time
   def to_json(json : JSON::Builder)
-    json.string(Time::Format::ISO_8601_DATE_TIME.format(self))
+    json.string(Time::Format::RFC_3339.format(self))
```

Comment:
Why this change if ISO_8601 is still defined?

Reply:
The difference is mostly negligible, but RFC 3339 is generally the better (stricter) format for this purpose. `ISO_8601_DATE_TIME` by default formats a offset as `HHmm` while RFC 3339 requires a colon: `HH:mm`

The JSON specification doesn't include a Date data type, so it's up to the implementation. JavaScript's `Date.toJSON()` actually is RFC 3339 compliant (`YYYY-MM-DDTHH:mm:ss.sssZ`).

Reply:
The references I found said ISO8601 is the format used in Javascript. I think we should stick to 8601 here. Actually been less stricter is better IMO since an API built on top of this would be less pedantic to the user.

* https://www.w3schools.com/jsref/jsref_tojson.asp
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString

Reply:
True, this is a bit confusing because RFC 3339 actually *is* ISO 8601. But RFC 3339 is better suited as machine-parsed exchange format. That's why I'm pretty convinced we should use RFC 3339 as output format for `#to_json`.

Reply:
I left `RFC_3339` here but it returns the exact same result as `ISO_8601_DATE_TIME`.